tests/suites/os/tests/secureboot/index.js

```diff
@@ -1,145 +1,193 @@
+				['ls', securebootEfiVarPath, '&>/dev/null',
+					'&&', 'echo', 'true',
+					'||', 'echo', 'false'
+				],
+				this.link,
+			).then(output => { return output === 'true'});
+	}
+
+	async isSecureBootEnabled() {
+			// https://wiki.archlinux.org/title/Unified_Extensible_Firmware_Interface/Secure_Boot#Checking_Secure_Boot_status
+			return this.worker.executeCommandInHostOS(
+					[
+						'od', '--address-radix=n', '--format=u1', securebootEfiVarPath,
+							'|', 'tr', '-s', "' '",
+							'|', 'cut', "-d' '", '-f6'
+					],
+					this.link,
+			).then(output => output === '1' );
+	}
+
+	async testBootloaderIntegrity() {
+		return this.worker.executeCommandInHostOS(
+			['balena', 'run', '--rm', '-v', '/mnt:/mnt', 'alpine', '/bin/sh', '-c',
+				'"apk add --update --no-cache binutils && strip /mnt/boot/EFI/BOOT/bootx64.efi"'],
+			this.link,
+		).then(() => this.worker.executeCommandInHostOS('reboot', this.link)
+		).then(() => this.test.resolves(
+			this.waitForSerialOutput(/bootx64.efi: Access Denied/),
+			'Firmware will not load bootloader that fails verification',
+		)
+		).then( () => this.resetDUT() );
+	};
+
+	async testBootloaderConfigIntegrity() {
+		return this.worker.executeCommandInHostOS(
+			['sed', '-i', 's/lockdown=integrity//', '/mnt/efi/EFI/BOOT/grub.cfg'],
+			this.link
+		).then(() => this.worker.executeCommandInHostOS('reboot', this.link)
+		).then(() => this.test.resolves(
+			/* The below pattern is the expected output when the config file
+			 * fails the signature check and the fallback console is not enabled.
+			 *
+			 * This sequence of ANSI escape codes decodes as:
+			 *
+			 *	ESC[0m - reset all modes
+			 *	ESC[37m - white foreground
+			 *	ESC[40m - black background
+			 *
+			 *	https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797
+			 *
+			 * Otherwise, the output finishes the above background color escape
+			 * sequence, clears the screen, and shows 'loading Boot0003
+			 * "balenaOS"'
+			 */
+
+			this.waitForSerialOutput(/\x1B\[0m\x1B\[37m\x1B\[40m$/, 3), // eslint-disable-line no-control-regex
+			'Bootloader will not load configuration that fails signature verification',
+			{ todo: 'This needs reworked for GRUB 2.12' },
+		)
+		).then( () => this.resetDUT() );
+	};
+}
+
+class testSecureBoot {
+	constructor(impl) {
+		this.impl = impl;
+	}
+
+	async run(test) {
+		if ( await this.impl.isSecureBootSupported() ) {
+			if ( await this.impl.isSecureBootEnabled() ) {
+				await this.impl.testFullDiskEncryption();
+				await this.impl.testModuleVerification();
+				await this.impl.testBootloaderIntegrity();
+				await this.impl.testBootloaderConfigIntegrity();
+			} else {
+				test.comment('Secure boot is not enabled');
+			}
+		} else {
+			test.comment('Secure boot is not supported by firmware');
+		}
+	}
+}
+
+module.exports = {
+	title: 'secure boot tests',
+	tests: [
+		{
+			title: 'check UEFI secure boot',
+			title: 'check secure boot',
```

Comment:
TAP JSON won't be happy accepting 2 keys of the same property `title`? Remove one of them